client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -181,7 +181,22 @@ class SimplePaymentsDialog extends Component {
 		return new Promise( resolve => this.formStateController.handleSubmit( resolve ) );
 	}
 
-	handleChangeTabs = activeTab => this.setState( { activeTab } );
+	handleChangeTabs = activeTab => {
+		if ( activeTab === 'form' ) {
+			this.editPaymentButton( null );
+		} else {
+			this.showButtonList();
+		}
+	};
+
+	showButtonList() {
+		this.setState( { activeTab: 'list' } );
+	}
+
+	editPaymentButton = editedPaymentId => {
```

Comment:
I'm renaming this to `showButtonForm`. It has a nice symmetry with `showButtonList`. Also, it's also used to show the "Add New" form, and `editPaymentButton` might suggest that it's only for editing existing button. Let me know what you think :)